### PR TITLE
Hide secret values by default

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Hide secret values by default using `input type="password"`. They can be shown and hidden using the eye icon, just like regular password inputs.

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

**Comment by jdunkerley** [comment]
Thank you :)

File: app/ide-desktop/lib/dashboard/src/modals/UpsertSecretModal.tsx [hunk 1]
```diff
@@ -80,16 +85,26 @@ export default function UpsertSecretModal(props: UpsertSecretModalProps) {
...
-            placeholder={
-              isNameEditable ? getText('secretValuePlaceholder') : getText('secretValueHidden')
-            }
-            className="text grow rounded-full border border-primary/10 bg-transparent px-input-x"
-            onInput={event => {
-              setValue(event.currentTarget.value)
-            }}
-          />
+          <div className="relative grow">
+            <input
```
Comment (MrFlashAccount) [inline app/ide-desktop/lib/dashboard/src/modals/UpsertSecretModal.tsx:89]: Hope eventually it'd a separate `<Password />` component

**Review by MrFlashAccount — APPROVED** [review]

File: app/ide-desktop/lib/dashboard/src/modals/UpsertSecretModal.tsx (hunk 1 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/modals/UpsertSecretModal.tsx:89]: yeah definitely. the issue is that we have multiple *slightly* different styles for the various areas (auth, settings, modals) and so we should *probably* figure out how to unify them first (or i suppose we can make the component styleable)

Comment (MrFlashAccount) [reply app/ide-desktop/lib/dashboard/src/modals/UpsertSecretModal.tsx:89]: It's technically possible using [`tw-merge`](https://www.npmjs.com/package/tailwind-merge) - we can override default styles in place, and tw-merge will dedupe classes

**Pull request merged**